var/spack/repos/builtin/packages/conduit/package.py

```diff
@@ -538,9 +538,9 @@ def hostconfig(self):
 
         cfg.write("# hdf5 from spack \n")
 
-        if "+hdf5" in spec:
+        if spec.satisfies("+hdf5"):
             cfg.write(cmake_cache_entry("HDF5_DIR", spec["hdf5"].prefix))
-            if "zlib-api" in spec:
+            if spec.satisfies("zlib-api"):
```

Comment:
This seems incorrect.
```suggestion
            if spec.satisfies("^zlib-api"):
```